### PR TITLE
Fix `compareBounds`

### PR DESCRIPTION
**Pull request by Alfred-Mountfield**
## 🌟 What is the purpose of this PR?

When translating the `compareBounds` implementation from Rust to TS we missed a few cases, this fixes a bug that occurs in an edge-case when the `kind` and `limit`s are the same, but the "type" (start or end) is different.

### CODE REVIEW

**Review by TimDiekmann — COMMENTED** [review]
The expression can be simplified, it just missed the case where both kinds are "inclusive".

File: libs/@blockprotocol/graph/src/stdlib/bound.ts [hunk 1]
```diff
@@ -14,7 +14,25 @@ export const compareBounds = (
...
+    (left.kind === "exclusive" &&
+      right.kind === "exclusive" &&
+      leftType === "start" &&
+      rightType === "start") ||
+    (left.kind === "exclusive" &&
+      right.kind === "exclusive" &&
+      leftType === "end" &&
+      rightType === "end") ||
+    (left.kind === "inclusive" && right.kind === "inclusive")
+  ) {
```
[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply libs/@blockprotocol/graph/src/stdlib/bound.ts:35]: Agreed, but last time we spoke about these we decided on keeping it as explicit about covering every edge-case as we could, and optimising once we add tests to this lib

[the remark this answers is not in the document]
Comment (TimDiekmann) [reply libs/@blockprotocol/graph/src/stdlib/bound.ts:35]: This is every edge case. 😆

**Review by TimDiekmann — APPROVED** [review]

File: libs/@blockprotocol/graph/src/stdlib/bound.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply libs/@blockprotocol/graph/src/stdlib/bound.ts:35]: > I don't ask to change this here as the corresponding Rust code is very similar but very well tested and writing the same tests again is a little bit too much. Ideally, I'd favor to have this functionality exported from a Rust crate (this applies to all interval-related operations).
 
- [comment](https://github.com/blockprotocol/blockprotocol/pull/924#discussion_r1092260547)

[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply libs/@blockprotocol/graph/src/stdlib/bound.ts:35]: > This is every edge case. 😆

_explicit_ 😁
